app/models/hotel.rb

```diff
@@ -1,3 +1,11 @@
 class Hotel < ApplicationRecord
   has_many :rooms
+
+
+  # def hotel_guests
```

Comment:
Something I like to think about when doing complex joins is starting LEFT and going RIGHT, where left is the model I'm starting at (Hotel) and Right is the table I'm trying to get to (guests). 
In this case, we can join hotels to rooms because of the one-to-many relationship, and then using the joins table under the hood, we can join rooms to guests. In ActiveRecord, we don't even need to include an explicit join with the joins table as AR will do that implicitly because you've set up relationships to allow guests to associate to rooms (`has_many :rooms, through :guest_rooms`). 
The AR query you could do here that works is:
`Hotel.joins(rooms: :guests).distinct("guests.name").pluck("guests.name")`

When I do that in rails console, the SQL given is:
`SELECT DISTINCT "guests"."name" FROM "hotels" INNER JOIN "rooms" ON "rooms"."hotel_id" = "hotels"."id" INNER JOIN "guest_rooms" ON "guest_rooms"."room_id" = "rooms"."id" INNER JOIN "guests" ON "guests"."id" = "guest_rooms"."guest_id"`
It looks complex, but it's just doing a chain of joins from left to right: hotels -> rooms -> guest_rooms -> guests

Another option that is way simpler is a little ActiveRecord magic. Because you have the one-to-many relationship between hotels and rooms, you could add this to the Hotel model: `has_many :guests, through: :rooms`
This will allow you to do `hotel.guests.distinct` and achieve the same result. We haven't talked about this much, but it's possible to shortcut the joining by adding this association, which associates another model to a given model if that other model is associated to the given model's child.